src/System.Private.CoreLib/shared/System/Diagnostics/DebugProvider.cs

```diff
@@ -81,11 +86,11 @@ internal static int IndentSize
             }
         }
 
-        private static bool s_needIndent;
+        private static bool s_needIndent = true;
```

Comment:
Does `s_needIndent` need to be threadstatic?  Even better - can it be an instance field instead of static?

Reply:
I can't see the rest of the code at the moment, but indent is usually thread static in debug tracing utility code as logically it belongs to one thread?

Reply:
This is always accesses under a global lock today, so it does not need to be thread static (noticed this after posting the comment).

Reply:
It may be nice slapping a comment on here noting this, just in case someone comes along and tries accessing it outside of the lock.

I still wonder if it would be better to make this an instance field though...